interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -489,23 +557,26 @@ def _get_cost_node(self, i, N_horizon, xtraj_node, utraj_node, ptraj_node, sl_no
                     ptraj_node[0],
                     sl_node[0],
                     su_node[0],
-                    ocp.get_initial_cost_expression(),
+                    self.get_cost_expression(ocp, ptraj_node[0][dims.np:dims.ny+dims.np], i),
```

Comment:
could you add an optional argument to the existing functions, `get_initial_cost_expression` and the like, for `y_ref` instead of adding `get_cost_expression`, which does not really belong to the `AcadosCasadi*` class IMO.